test/parallel/test-runner-mocking.js

```diff
@@ -434,6 +434,31 @@ test('spy functions can be used on classes inheritance', (t) => {
   assert.strictEqual(C.someTask.mock, undefined);
 });
 
+test('spy functions don\'t affect the prototype chain', (t) => {
+
+  class A {
+    static someTask(msg) {
+      return msg;
+    }
+  }
+  class B extends A {}
+  class C extends B {}
+
+  const msg = 'ok';
+
+  t.mock.method(C, C.someTask.name);
+  C.someTask(msg);
+
+  assert.strictEqual(B.someTask.mock, undefined);
+  assert.strictEqual(A.someTask.mock, undefined);
```

Comment:
why would A need to have the method removed? I’d expect it to still be there - you mocked C, not A. Also, shouldn’t this be checking A.someTask directly?

Reply:
I think I didn't get it. The method wasn't removed nor added to A or B. The mock happened only on C and any restore would only affect C. This was the test I wrote. Did you think about something else?

Reply:
maybe i'm misunderstanding the assertions here. I don't think we should be checking for the `.mock` property, which is an implementation detail of the mocking code - i think they should be asserting that the descriptor for `A.someTask` is unchanged, that `!('someTask' in B)`, and that the descriptor for `C.someTask` now exists. Does that clarify?

Reply:
Nice! Just updated it. Would you check it?